src/update.rs

```diff
@@ -203,10 +312,136 @@ fn handle_toml_file(path: PathBuf, rewrite: &Rewrite, version: &Version) -> Resu
                     let table = toml_doc[k][dn]
                         .as_inline_table_mut()
                         .expect("We filter by `is_inline_table`; qed");
-                    handle_dependency(dn, table, rewrite, version);
+                    let _ = handle_dependency(dn, table, rewrite, key).map_err(|err| {
+                        log::error!("Error handling dependency: {}", err);
+                    });
                 })
         });
 
     fs::write(&path, toml_doc.to_string())?;
     Ok(())
 }
+
+/// Get the source of where to get the versions from.
+fn get_version_source(version: &String) -> Result<VersionSource> {
+    let source = if version.starts_with("http://") || version.starts_with("https://") {
+        VersionSource::Url(version.clone())
+    } else {
+        let path = PathBuf::from(version);
+        if path.exists() && path.file_name() == Some("Cargo.lock".as_ref()) {
+            VersionSource::File(version.clone())
+        } else if version == "latest" {
+            VersionSource::CratesIO
+        } else {
+            bail!("Invalid 'version' source: {}", version)
+        }
+    };
+    Ok(source)
+}
+
+/// Get the version of a package from a given source.
+fn get_version_by_source(package: &str, source: &VersionSource) -> Result<String> {
+    let version = match source {
+        VersionSource::CratesIO => {
+            let url = format!("https://crates.io/api/v1/crates/{}", package);
```

Comment:
Are you sure that we don't generate too many requests? Maybe we should wait between requests a little bit.

Reply:
Yes it has limits. I used https://crates.io/data-access which is unlimited [here
](https://github.com/ggwpez/substrate-scripts/blob/4ba5b0df9db832eaecd8d1b507dfc75322c3c795/import-runtime-repos/fix-deps/src/main.rs#L23-L28).  
The create name needs to be split up, but it is not rate-limited as advantage.

Reply:
> Are you sure that we don't generate too many requests? Maybe we should wait between requests a little bit.

I tested it and it works. Maybe thanks to `reqwest::blocking`?